modules/ROOT/pages/traversal-framework/traversal_framework_example.adoc

```diff
@@ -0,0 +1,210 @@
+:description: The Neo4j Traversal Framework Example.
+
+:org-neo4j-graphdb-Direction-both: {neo4j-javadocs-base-uri}/org/neo4j/graphdb/Direction.html#BOTH
+
+[[examples-how-to-use-the-traversal-framework]]
+= Traversal Framework Example
+
+A link:{org-neo4j-graphdb-traversal-TraversalDescription}[`org.neo4j.graphdb.traversal.TraversalDescription`^] is built using a fluent interface and such a description can then spawn several link:{org-neo4j-graphdb-traversal-Traverser}[`org.neo4j.graphdb.traversal.Traverser`^] objects.
+
+image::traversal_framework_example.svg[role="middle"]
+
+[NOTE]
+====
+The source code for the examples can be found here: link:https://github.com/neo4j/neo4j-documentation/blob/{neo4j-version}/embedded-examples/src/main/java/org/neo4j/examples/TraversalExample.java[`TraversalExample.java`^].
+====
+
+With the definition of the RelationshipTypes as:
+
+[source, java]
+----
+private enum Rels implements RelationshipType
+{
+    LIKES, KNOWS
+}
+----
+
+The graph can be traversed with for example the following traverser, starting at the node with the `name = 'Joe'`:
+
+[source, java]
+----
+for ( Path position : db.traversalDescription()
+        .depthFirst()
+        .relationships( Rels.KNOWS )
+        .relationships( Rels.LIKES, Direction.INCOMING )
+        .evaluator( Evaluators.toDepth( 5 ) )
+        .traverse( node ) )
+{
+    output += position + "\n";
+}
+----
+
+The traversal will output:
+
+[source, output, role="noheader"]
+----
+(0)
+(0)<-[LIKES,1]-(5)
+(0)<-[LIKES,1]-(5)-[KNOWS,6]->(1)
+(0)<-[LIKES,1]-(5)-[KNOWS,6]->(1)<-[KNOWS,5]-(6)
+(0)<-[LIKES,1]-(5)-[KNOWS,6]->(1)-[KNOWS,4]->(4)
+(0)<-[LIKES,1]-(5)-[KNOWS,6]->(1)-[KNOWS,4]->(4)-[KNOWS,3]->(3)
+(0)<-[LIKES,1]-(5)-[KNOWS,6]->(1)-[KNOWS,4]->(4)-[KNOWS,3]->(3)-[KNOWS,2]->(2)
+----
+
+Since a `TraversalDescription` is immutable it is also useful to create template descriptions which holds common settings shared by different traversals.
```

Comment:
```suggestion
Since a `TraversalDescription` is immutable, it is useful to create template descriptions that include common settings shared by different traversals.
```